files/usr/share/cinnamon/applets/cornerbar@cinnamon.org/applet.js

```diff
@@ -154,6 +155,17 @@ class CinnamonBarApplet extends Applet.Applet {
         }
     }
 
+    _on_scroll_event(actor, event) {
+        //switch workspace
+        if (this._did_peek == true && this._peek_timeout_id > 0) {
+            Mainloop.source_remove(this._peek_timeout_id);
+        }
+        var index = global.screen.get_active_workspace_index() + event.get_scroll_direction() * 2 - 1;
```

Comment:
This doesn't make sense:   `index + (scroll direction * 2)  - 1`

A couple of things:
- In clutter, you get 2 scroll events per 'real' event - one with normal Clutter.ScrollDirection.UP/DOWN/LEFT/RIGHT (0, 1, 2, 3), and one with Clutter.ScrollDirection.SMOOTH (4). You need to choose to listen to one or the other, and filter the others out before acting on anything. Generally we always use up/down/etc... because we're only interested in direction, not distance/velocity. So you'd do something like [this](https://github.com/linuxmint/Cinnamon/blob/master/js/ui/popupMenu.js#L798-L802) or check for specific directions only and ignore the rest.
- Also (only a suggestion), you should just operate plainly when possible, for determining which index to use, instead of some formula like this - use the ScrollDirection enums to keep things readable:
```
if (scroll_direction === Clutter.ScrollDirection.UP) {
    index = ......
}
...
...